mesa/testProton.py

```diff
@@ -55,17 +55,16 @@ def test_generate_households():
         other_members = set([x for x in test_family if x != member])
         assert other_members == member.neighbors["household"]
     #3
-    for test_simple_family in m.families[:3]:
-        if len(test_simple_family) >= 3:
-            break
-    assert set(test_simple_family[1:]) == test_simple_family[0].neighbors["household"]
-    assert test_simple_family[-1] in test_simple_family[0].neighbors["offspring"]
-    assert test_simple_family[-1] in test_simple_family[1].neighbors["offspring"]
-    if type(test_simple_family[2:]) == list:
-        for son in test_simple_family[2:]:
-            assert son.neighbors["parent"] == set(test_simple_family[:2])
-    else:
-        assert test_simple_family[-1].neighbors["parent"] == set(test_simple_family[:2])
+    for test_simple_family in m.families[:5]:
```

Comment:
Why are we testing the first five only?

Reply:
The generation takes place in two phases. In the first phase all "simple" families are generated. By simple families we mean those that meet the parameters that come from the statistical tables. (These first families are also the largest). In the second phase, on the other hand, complex families are generated, which do not properly respect the statistical parameters. Taking the first 5 we are sure that these are simple families. I will add a way for all the simple families to be tested.